osu.Game/Screens/Play/PlayerSettings/BeatmapOffsetControl.cs

```diff
@@ -277,6 +284,9 @@ protected override void Dispose(bool isDisposing)
         protected override void Update()
         {
             base.Update();
+
+            if (useAverageButton != null)
+                useAverageButton.Enabled.Value = allowOffsetAdjust;
```

Comment:
`useAverageButton.Enabled` is touched in another place using a completely different conditional:

https://github.com/ppy/osu/blob/4ae6dfd790470276c90f0f295993bc85f49e4704/osu.Game/Screens/Play/PlayerSettings/BeatmapOffsetControl.cs#L156-L159

What that means is that after clicking the "use average" button it won't disable like it would previously. Maybe that's fine to have, but in that case I'd just remove that other code because why have it there at that point if it's gonna get instantly overridden. (But the best of both worlds situation would probably be to use a conjunction of both conditionals in both places.)

Reply:
Good catch 😅 

Have pushed another fix for this which accounts for both flags.